src/Core/Geometry/MeshPrimitives.inl

```diff
@@ -1,98 +1,202 @@
 #pragma once
 #include <Core/Geometry/MeshPrimitives.hpp>
+#include <Core/Geometry/TopologicalMesh.hpp>
 #include <Core/Math/LinearAlgebra.hpp> // Math::pi
 
+#include <random>
+
 namespace Ra {
 namespace Core {
 namespace Geometry {
 
 template <uint U, uint V>
-TriangleMesh makeParametricSphere( Scalar radius, const Utils::optional<Utils::Color>& color ) {
+TriangleMesh
+makeParametricSphere( Scalar radius, const Utils::optional<Utils::Color>& color, bool gtc ) {
     constexpr uint slices = U;
     constexpr uint stacks = V;
 
-    TriangleMesh result;
-
-    TriangleMesh::PointAttribHandle::Container vertices;
-    TriangleMesh::NormalAttribHandle::Container normals;
-    TriangleMesh::IndexContainerType indices;
-    Ra::Core::Vector3Array texCoords;
-
-    vertices.reserve( 2 + slices * ( stacks - 1 ) );
-    normals.reserve( 2 + slices * ( stacks - 1 ) );
-    indices.reserve( 2 * slices * ( stacks - 1 ) );
-
     const Scalar du = 1_ra / slices;
     const Scalar dv = 1_ra / stacks;
-
+    using WAI       = TopologicalMesh::WedgeAttribIndex;
+    TopologicalMesh topoMesh;
+    auto whNormal = topoMesh.addWedgeAttrib<Vector3>( getAttribName( MeshAttrib::VERTEX_NORMAL ) );
+
+    WAI whColor =
+        color ? topoMesh.addWedgeAttrib<Vector4>( getAttribName( MeshAttrib::VERTEX_COLOR ) )
+              : WAI::Invalid();
+    WAI whTexCoord =
+        gtc ? topoMesh.addWedgeAttrib<Vector3>( getAttribName( MeshAttrib::VERTEX_TEXCOORD ) )
+            : WAI::Invalid();
+
+    TopologicalMesh::VertexHandle vhandles[( stacks - 1 ) * slices + 2];
+    Vector3Array topoTexCoords;
+    topoTexCoords.reserve( ( stacks - 1 ) * slices + 2 + 2 * slices );
```

Comment:
Why is there a different number of tex coord than number of vertices/normals ?

Reply:
for pole I think

Reply:
thinking again, it's for seams where u=1 meets u=0